radix-engine/src/engine/track.rs

```diff
@@ -12,10 +14,31 @@ use crate::errors::RuntimeError;
 use crate::ledger::*;
 use crate::model::*;
 
+
+// TODO: Replace NonFungible with real re address
+// TODO: Move this logic into application layer
+macro_rules! resource_to_non_fungible_space {
+    ($resource_address:expr) => {{
+        let mut addr = scrypto_encode(&$resource_address);
+        addr.push(0u8);
+        addr
+    }};
+}
```

Comment:
Sorry, understanding you have a `TODO` here, why do we need conversion?

Reply:
This is simply to derive a non_fungible_space address given a resource_address